app/code/Magento/Catalog/Model/Indexer/Product/Flat/Action/Eraser.php

```diff
@@ -85,13 +85,17 @@ public function removeDisabledProducts(array &$ids, $storeId)
         $select->joinLeft(
             ['status_global_attr' => $statusAttribute->getBackendTable()],
             ' status_global_attr.attribute_id = ' . (int)$statusAttribute->getAttributeId()
-            . ' AND status_global_attr.store_id = ' . Store::DEFAULT_STORE_ID,
+            . ' AND status_global_attr.store_id = ' . Store::DEFAULT_STORE_ID
+            . ' AND status_global_attr.' . $statusAttribute->getEntityIdField() . '='
+            . 'product_table.' . $statusAttribute->getEntityIdField(),
```

Comment:
I guess we should not use the `$statusAttribute->getEntityIdField()` method's call in order to get the link field name of `catalog_product_entity` DB table. This is rather a coincidence that linked fields have the same names. Please use the same approach as we already have in `\Magento\Catalog\Model\Indexer\Product\Flat\Action\Eraser::getSelectForProducts`.

Reply:
Could you please elaborate? I do not see anything in the [referenced function](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Catalog/Model/Indexer/Product/Flat/Action/Eraser.php#L118-L125).

While I do agree that fetching entity id field from `$statusAttribute` is wrong, wouldn't proper way than be to get `Magento\Catalog\Api\Data\ProductInterface` metadata from `Magento\Framework\EntityManager\MetadataPool` and call `getLinkField()` on it?

Reply:
I mean, that using the hardcoded field name `product_table.entity_id` is just enough in this case.

Reply:
I would recommend removing MetadataPool dependency.

Reply:
This pull request is coming from bugfix I did for our EE client. Hardcoding `entity_id` there would cause incorrect relation as EAV attributes are joined by `row_id` instead. Hence the usage of MetadataPool to make it compatible for both.

Correct me if I am missing something.

Reply:
Thank you for the update!
Agreed.